numba/core/base.py

```diff
@@ -431,7 +431,8 @@ def declare_function(self, module, fndesc):
         fn = cgutils.get_or_insert_function(module, fnty, fndesc.mangled_name)
         self.call_conv.decorate_function(fn, fndesc.args, fndesc.argtypes, noalias=fndesc.noalias)
         if fndesc.inline:
-            fn.attributes.add('alwaysinline')
+            if 'noinline' not in fn.attributes:
+                fn.attributes.add('alwaysinline')
```

Comment:
How does this situation occur? Does it imply that if a function is described with `inline` but has attr `noinline` it will remain so even if a user specifies `forceinline`, this state seems a bit confusing?

Reply:
I changed the logic. Now `forceinline` will always override `optnone`.